tests/qnodes/test_qnode_reversible.py

```diff
@@ -0,0 +1,849 @@
+        assert gradA == pytest.approx(expected, abs=tol)
+
+    @pytest.mark.xfail
+    def test_controlled_RY_gradient(self, tol):
+        """Test gradient of controlled RY gate"""
+        dev = qml.device("default.qubit", wires=2)
+
+        def circuit(x):
+            qml.PauliX(wires=0)
+            qml.CRY(x, wires=[0, 1])
+            return qml.expval(qml.PauliZ(0))
+
+        circuit = ReversibleQNode(circuit, dev)
+
+        a = 0.542  # any value of a should give zero gradient
+
+        # get the analytic gradient
+        gradA = circuit.jacobian([a], method="A")
+        # get the finite difference gradient
+        gradF = circuit.jacobian([a], method="F")
+
+        # the expected gradient
+        expected = 0
+
+        assert gradF == pytest.approx(expected, abs=tol)
+        assert gradA == pytest.approx(expected, abs=tol)
+
+        def circuit1(x):
+            qml.RX(x, wires=0)
+            qml.CRY(x, wires=[0, 1])
+            return qml.expval(qml.PauliZ(0))
+
+        circuit1 = ReversibleQNode(circuit1, dev)
+
+        b = 0.123  # gradient is -sin(x)
+
+        # get the analytic gradient
+        gradA = circuit1.jacobian([b], method="A")
+        # get the finite difference gradient
+        gradF = circuit1.jacobian([b], method="F")
+
+        # the expected gradient
+        expected = -np.sin(b)
+
+        assert gradF == pytest.approx(expected, abs=tol)
+        assert gradA == pytest.approx(expected, abs=tol)
+
+    @pytest.mark.xfail
+    def test_controlled_RZ_gradient(self, tol):
+        """Test gradient of controlled RZ gate"""
+        dev = qml.device("default.qubit", wires=2)
+
+        def circuit(x):
+            qml.PauliX(wires=0)
+            qml.CRZ(x, wires=[0, 1])
+            return qml.expval(qml.PauliZ(0))
+
+        circuit = ReversibleQNode(circuit, dev)
+
+        a = 0.542  # any value of a should give zero gradient
+
+        # get the analytic gradient
+        gradA = circuit.jacobian([a], method="A")
+        # get the finite difference gradient
+        gradF = circuit.jacobian([a], method="F")
+
+        # the expected gradient
+        expected = 0
+
+        assert gradF == pytest.approx(expected, abs=tol)
+        assert gradA == pytest.approx(expected, abs=tol)
+
+        def circuit1(x):
+            qml.RX(x, wires=0)
+            qml.CRZ(x, wires=[0, 1])
+            return qml.expval(qml.PauliZ(0))
+
+        circuit1 = ReversibleQNode(circuit1, dev)
+
+        b = 0.123  # gradient is -sin(x)
+
+        # get the analytic gradient
+        gradA = circuit1.jacobian([b], method="A")
+        # get the finite difference gradient
+        gradF = circuit1.jacobian([b], method="F")
+
+        # the expected gradient
+        expected = -np.sin(b)
+
+        assert gradF == pytest.approx(expected, abs=tol)
+        assert gradA == pytest.approx(expected, abs=tol)
+
+
+class TestVarianceJacobian:
+    """Variance analytic jacobian integration tests."""
```

Comment:
Since `_pd_analytic_var` is not defined for the `ReversibleQNode`, and is just inherited from `QubitQNode`, these tests should be removed as well.